hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/health/GatewayBaselineProviderImpl.java

```diff
@@ -0,0 +1,146 @@
+import org.hypertrace.graphql.metric.request.Baseline;
+import org.hypertrace.graphql.metric.request.MetricRequest;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@Singleton
+public class GatewayBaselineProviderImpl implements BaselineProvider {
+  private static final int DEFAULT_DEADLINE_SEC = 10;
+  private final GatewayServiceEntityRequestBuilder gatewayServiceEntityRequestBuilder;
+  private GatewayServiceGrpc.GatewayServiceFutureStub gatewayServiceStub;
+  private final GraphQlGrpcContextBuilder grpcContextBuilder;
+
+  @Inject
+  GatewayBaselineProviderImpl(
+      GatewayServiceEntityRequestBuilder gatewayServiceEntityRequestBuilder,
+      GrpcChannelRegistry grpcChannelRegistry,
+      GraphQlServiceConfig serviceConfig,
+      CallCredentials credentials,
+      GraphQlGrpcContextBuilder grpcContextBuilder) {
+    this.gatewayServiceEntityRequestBuilder = gatewayServiceEntityRequestBuilder;
+    this.grpcContextBuilder = grpcContextBuilder;
+    this.gatewayServiceStub =
+        GatewayServiceGrpc.newFutureStub(
+                grpcChannelRegistry.forAddress(
+                    serviceConfig.getGatewayServiceHost(), serviceConfig.getGatewayServicePort()))
+            .withCallCredentials(credentials);
+  }
+
+  @Override
+  public Single<BaselineEntitiesRequest> buildRequest(
+      EntitiesRequest entitiesRequest, EntitiesResponse entitiesResponse, EntityRequest request) {
+    Map<String, Baseline> baselineRequestMap = getBaselineRequestMap(request);
+    BaselineEntitiesRequest baselineEntitiesRequest =
+        BaselineEntitiesRequest.newBuilder()
+            .setStartTimeMillis(entitiesRequest.getStartTimeMillis())
+            .setEndTimeMillis(entitiesRequest.getEndTimeMillis())
+            .addAllEntityIds(
+                entitiesResponse.getEntityList().stream()
+                    .map(x -> x.getId())
+                    .collect(Collectors.toList()))
+            .setEntityType(entitiesRequest.getEntityType())
+            .addAllBaselineAggregateRequest(
+                getFunctionExpression(entitiesRequest, baselineRequestMap))
+            .addAllBaselineMetricSeriesRequest(
+                getBaselineTimeSeriesRequest(entitiesRequest, baselineRequestMap))
+            .build();
+    return Single.just(baselineEntitiesRequest);
+  }
+
+  private Map<String, Baseline> getBaselineRequestMap(EntityRequest request) {
+    Map<String, Baseline> baselineMap = new HashMap<>();
+    List<MetricRequest> metricRequests = request.metricRequests();
+    if (metricRequests.size() > 0) {
+      for (MetricRequest metricRequest : metricRequests) {
+        baselineMap.putAll(
+            metricRequest.aggregationRequests().stream()
+                .filter(x -> x.baseline() != null)
```

Comment:
if baseline() returned a boolean as suggested elsewhere, this coded would also simplify. Are there any places we're using it beyond a null check?